homeassistant/components/hydrawise/switch.py

```diff
@@ -76,58 +78,43 @@ async def async_setup_entry(
     coordinator: HydrawiseDataUpdateCoordinator = hass.data[DOMAIN][
         config_entry.entry_id
     ]
-    default_watering_timer = DEFAULT_WATERING_TIME
-
-    entities = [
-        HydrawiseSwitch(
-            data=zone,
-            coordinator=coordinator,
-            description=description,
-            default_watering_timer=default_watering_timer,
-        )
-        for zone in coordinator.api.relays
-        for description in SWITCH_TYPES
-    ]
-
-    async_add_entities(entities)
+    async_add_entities(
+        [
+            HydrawiseSwitch(coordinator, description, controller, zone)
+            for controller in coordinator.data.controllers
+            for zone in controller.zones
+            for description in SWITCH_TYPES
+        ]
+    )
 
 
 class HydrawiseSwitch(HydrawiseEntity, SwitchEntity):
     """A switch implementation for Hydrawise device."""
 
-    def __init__(
-        self,
-        *,
-        data: dict[str, Any],
-        coordinator: HydrawiseDataUpdateCoordinator,
-        description: SwitchEntityDescription,
-        default_watering_timer: int,
-    ) -> None:
-        """Initialize a switch for Hydrawise device."""
-        super().__init__(data=data, coordinator=coordinator, description=description)
-        self._default_watering_timer = default_watering_timer
-
-    def turn_on(self, **kwargs: Any) -> None:
+    async def async_turn_on(self, **kwargs: Any) -> None:
```

Comment:
```suggestion
    zone: Zone

    async def async_turn_on(self, **kwargs: Any) -> None:
```
Maybe give this an attempt to get rid of the asserts

Reply:
Yep, that worked. Thanks!